Client/Frontend/Menu/AppMenuConfiguration.swift

```diff
@@ -112,7 +112,14 @@ struct AppMenuConfiguration: MenuConfiguration {
             if #available(iOS 9, *) {
                 menuItems.append(AppMenuConfiguration.NewPrivateTabMenuItem)
             }
-            menuItems.append(tabState.isBookmarked ? AppMenuConfiguration.RemoveBookmarkMenuItem : AppMenuConfiguration.AddBookmarkMenuItem)
+            var menuItem: MenuItem
+            if let errorPageUrl = tabState.url where ErrorPageHelper.isErrorPageURL(errorPageUrl) {
+                menuItem = AppMenuConfiguration.AddBookmarkMenuItem
+                menuItem.isDisabled = true
+            } else {
+                menuItem = tabState.isBookmarked ? AppMenuConfiguration.RemoveBookmarkMenuItem : AppMenuConfiguration.AddBookmarkMenuItem
+            }
+            menuItems.append(menuItem)
```

Comment:
I think this can be simplified a bit:
```
let menuItem = menuItem = tabState.isBookmarked ? AppMenuConfiguration.RemoveBookmarkMenuItem : AppMenuConfiguration.AddBookmarkMenuItem
if let errorPageUrl = tabState.url where ErrorPageHelper.isErrorPageURL(errorPageUrl) {
    menuItem.isDisabled = true
}
menuItems.append(menuItem)
```